src/index.js

```diff
@@ -90,28 +90,35 @@ function visitEntity(entity, entitySchema, bag, options) {
   }
 
   let stored = bag[entityKey][id];
-  let normalized = visitObject(entity, entitySchema, bag, options);
+  let normalized = visitObject(entity, entitySchema, bag, options, collectionKey);
   mergeIntoEntity(stored, normalized, entityKey);
 
   return id;
 }
 
-function visit(obj, schema, bag, options) {
+function visit(obj, schema, bag, options, collectionKey) {
   if (!isObject(obj) || !isObject(schema)) {
     return obj;
   }
 
   if (schema instanceof EntitySchema) {
-    return visitEntity(obj, schema, bag, options);
+    return visitEntity(obj, schema, bag, options, collectionKey);
   } else if (schema instanceof IterableSchema) {
     return visitIterable(obj, schema, bag, options);
   } else if (schema instanceof UnionSchema) {
     return visitUnion(obj, schema, bag, options);
   } else {
-    return visitObject(obj, schema, bag, options);
+    return visitObject(obj, schema, bag, options, collectionKey);
   }
 }
 
+function normalizeResult(result) {
```

Comment:
Result comes back as an object because the new functionality is added to an iterable schema (`arrayOf(foo)` or `keyedObjectOf(foo)`, as mentioned above). Thus, `visit()` returns the result of `visitIterable()`, which is an object mapping keys to the result of `curriedItemMapper(obj[key], key)`. 

When we use the function `(obj, key) => key`, this ultimately returns an object something like `{ 1: '1', 2: '2', 3: '3' }` (whereas something like `obj => obj` would return `{ 1: foo, 2: bar, 3: baz }` or something). Here, the values of the object don't represent objects, so we convert into an array of keys after double checking that all the keys do equal all their values.